### PR TITLE
Added documentation for dnsMessage option

### PR DESCRIPTION
**Pull request by BenjaminPaap**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.7 |
| Fixed tickets | #5802 |

### CODE REVIEW

File: reference/constraints/Url.rst [hunk 1]
```diff
@@ -10,6 +10,7 @@ Validates that a value is a valid URL string.
 |                | - `protocols`_                                                      |
 |                | - `payload`_                                                        |
 |                | - `checkDNS`_                                                       |
+|                | - `dnsMessage`_
```
Comment (xabbuh) [inline reference/constraints/Url.rst]: We need a `|` character in this line to end the cell (at the same position as on the other lines).

File: reference/constraints/Url.rst [hunk 2]
```diff
@@ -299,5 +300,78 @@ option to ``true``:
...
+             protected $bioUrl;
+        }
+
+    .. code-block:: yaml
+
+        # src/AppBundle/Resources/config/validation.yml
+        AppBundle\Entity\Author:
+            properties:
+                bioUrl:
+                    - Url: { dnsMessage: The host "{{ value }}" could not be resolved }
```
Comment (xabbuh) [inline reference/constraints/Url.rst]: We have to enclose the message in single quotes to have valid YAML here.

File: reference/constraints/Url.rst [hunk 3]
```diff
@@ -299,5 +300,78 @@ option to ``true``:
...
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('bioUrl', new Assert\Url(array(
+                     'dnsMessage' => 'The host "{{ value }}" could not be resolved'
+                )));
+            }
+        }
+
 This option uses the :phpfunction:`checkdnsrr` PHP function to check the validity
 of the ``ANY`` DNS record corresponding to the host associated with the given URL.
```
Comment (xabbuh) [inline reference/constraints/Url.rst]: This paragraph belongs to the previously described `checkDns` option.

**Comment by xabbuh** [comment]
Thank you @BenjaminPaap. The changes look very good. I just left you some minor comments. After applying them, this should be ready to be merged.

File: reference/constraints/Url.rst (hunk 2 above)
Comment (xabbuh) [inline reference/constraints/Url.rst]: Ah, and what do you think to end the message with a period (in all examples) to have a complete sentence?

File: reference/constraints/Url.rst [hunk 4]
```diff
@@ -301,3 +302,76 @@ option to ``true``:
...
+
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class Author
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('bioUrl', new Assert\Url(array(
+                     'dnsMessage' => 'The host "{{ value }}" could not be resolved.'
```
Comment (xabbuh) [inline reference/constraints/Url.rst]: We usually add a comma after each array entry.

**Comment by xabbuh** [comment]
:+1:

File: reference/constraints/Url.rst (hunk 4 above)
Comment (BenjaminPaap) [inline reference/constraints/Url.rst]: Good to know. Am I supposed to add one on this pr or just keeping this in mind for upcoming pr's?

Comment (xabbuh) [inline reference/constraints/Url.rst]: If you want to, you can make that change. Otherwise we will just apply this when merging. :)

**Comment by OskarStark** [comment]
great, thank you @BenjaminPaap :+1:

**Comment by weaverryan** [comment]
This looks perfect. Thanks Benjamin!

**Pull request merged**